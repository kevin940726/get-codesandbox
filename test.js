const fetch = require('node-fetch');
const path = require('path');
const fs = require('fs');
const { getCodeSandbox, uploadSandbox } = require('./');

// Some E2E tests need more time
jest.setTimeout(10000);

test('Official sandboxes', async () => {
  const { files, entry } = await getCodeSandbox('vanilla');

  expect({
    files,
    entry,
  }).toMatchSnapshot();
});

test('Sandbox ID', async () => {
  const { files, entry } = await getCodeSandbox('rjk9n4zj7m'); // Static

  expect({
    files,
    entry,
  }).toMatchSnapshot();
});

test('Github path', async () => {
  const { files, entry } = await getCodeSandbox(
    'github/codesandbox-app/static-template'
  );

  expect({
    files,
    entry,
  }).toMatchSnapshot();
});

test('File path', async () => {
  const { files, entry } = await getCodeSandbox('file:./examples/console');

  expect({ files, entry }).toMatchSnapshot();
});

test('File path with base path', async () => {
  const { files, entry } = await getCodeSandbox('file:./console', {
    basePath: path.resolve(__dirname, 'examples'),
  });

  expect({ files, entry }).toMatchSnapshot();
});

test('Upload binary files', async () => {
  const { files } = await getCodeSandbox('file:./examples/binary');
  const sandboxID = await uploadSandbox({ files });

  const received = await fetch(
    `https://${sandboxID}.csb.app/src/cat.png`
  ).then((res) => res.buffer());

  const expected = await fs.promises.readFile('examples/binary/src/cat.png');

  expect(received.equals(expected)).toBe(true);
});
