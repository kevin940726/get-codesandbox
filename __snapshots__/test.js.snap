// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File path 1`] = `
Object {
  "entry": "index.html",
  "files": Object {
    "index.ts": Object {
      "content": "console.log(\\"Hello get-codesandbox!\\");
",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"console\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.ts\\",
  \\"scripts\\": {
    \\"start\\": \\"parcel index.html --open\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"7.2.0\\",
    \\"parcel-bundler\\": \\"^1.6.1\\"
  },
  \\"keywords\\": []
}
",
    },
  },
}
`;

exports[`File path with base path 1`] = `
Object {
  "entry": "index.html",
  "files": Object {
    "index.ts": Object {
      "content": "console.log(\\"Hello get-codesandbox!\\");
",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"console\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"main\\": \\"index.ts\\",
  \\"scripts\\": {
    \\"start\\": \\"parcel index.html --open\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"7.2.0\\",
    \\"parcel-bundler\\": \\"^1.6.1\\"
  },
  \\"keywords\\": []
}
",
    },
  },
}
`;

exports[`Github path 1`] = `
Object {
  "entry": "index.html",
  "files": Object {
    "index.html": Object {
      "content": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Static Template</title>
  </head>
  <body>
    <h1>This is a static template, there is no bundler or bundling involved!</h1>
  </body>
</html>",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"static\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"This is a static template with no bundling\\",
  \\"main\\": \\"index.html\\",
  \\"scripts\\": {
    \\"start\\": \\"serve\\",
    \\"build\\": \\"echo This is a static template, there is no bundler or bundling involved!\\"
  },
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+https://github.com/codesandbox-app/static-template.git\\"
  },
  \\"keywords\\": [
    \\"static\\",
    \\"template\\",
    \\"codesandbox\\"
  ],
  \\"author\\": \\"Ives van Hoorne\\",
  \\"license\\": \\"MIT\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/codesandbox-app/static-template/issues\\"
  },
  \\"homepage\\": \\"https://github.com/codesandbox-app/static-template#readme\\",
  \\"devDependencies\\": {
    \\"serve\\": \\"^11.2.0\\"
  }
}
",
    },
    "sandbox.config.json": Object {
      "content": "{
  \\"template\\": \\"static\\"
}
",
    },
  },
}
`;

exports[`Official sandboxes 1`] = `
Object {
  "entry": "src/index.js",
  "files": Object {
    "index.html": Object {
      "content": "<!DOCTYPE html>
<html>

<head>
	<title>Parcel Sandbox</title>
	<meta charset=\\"UTF-8\\" />
</head>

<body>
	<div id=\\"app\\"></div>

	<script src=\\"src/index.js\\">
	</script>
</body>

</html>",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"vanilla\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"JavaScript example starter project\\",
  \\"main\\": \\"index.html\\",
  \\"scripts\\": {
    \\"start\\": \\"parcel index.html --open\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"7.2.0\\",
    \\"parcel-bundler\\": \\"^1.6.1\\"
  },
  \\"keywords\\": [
    \\"javascript\\",
    \\"starter\\"
  ]
}",
    },
    "src/index.js": Object {
      "content": "import \\"./styles.css\\";

document.getElementById(\\"app\\").innerHTML = \`
<h1>Hello Vanilla!</h1>
<div>
  We use the same configuration as Parcel to bundle this sandbox, you can find more
  info about Parcel 
  <a href=\\"https://parceljs.org\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">here</a>.
</div>
\`;
",
    },
    "src/styles.css": Object {
      "content": "body {
  font-family: sans-serif;
}
",
    },
  },
}
`;

exports[`Sandbox ID 1`] = `
Object {
  "entry": "index.html",
  "files": Object {
    "index.html": Object {
      "content": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Static Template</title>
  </head>
  <body>
    <h1>This is a static template, there is no bundler or bundling involved!</h1>
  </body>
</html>",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"static\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"This is a static template with no bundling\\",
  \\"main\\": \\"index.html\\",
  \\"scripts\\": {
    \\"start\\": \\"serve\\",
    \\"build\\": \\"echo This is a static template, there is no bundler or bundling involved!\\"
  },
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+https://github.com/codesandbox-app/static-template.git\\"
  },
  \\"keywords\\": [
    \\"static\\",
    \\"template\\",
    \\"codesandbox\\"
  ],
  \\"author\\": \\"Ives van Hoorne\\",
  \\"license\\": \\"MIT\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/codesandbox-app/static-template/issues\\"
  },
  \\"homepage\\": \\"https://github.com/codesandbox-app/static-template#readme\\",
  \\"devDependencies\\": {
    \\"serve\\": \\"^11.2.0\\"
  }
}
",
    },
    "sandbox.config.json": Object {
      "content": "{
  \\"template\\": \\"static\\"
}
",
    },
  },
}
`;
